app/src/main/java/org/openobservatory/ooniprobe/test/suite/ExperimentalSuite.java

```diff
@@ -26,14 +27,34 @@ public ExperimentalSuite() {
                 R.string.TestResults_NotAvailable);
     }
 
-    @Override public AbstractTest[] getTestList(@Nullable PreferenceManager pm) {
+    public static ExperimentalSuite initForAutoRun() {
+        ExperimentalSuite suite = new ExperimentalSuite();
+        suite.setAutoRun(true);
+        return suite;
+    }
+
+
+    @Override
+    public AbstractTest[] getTestList(@Nullable PreferenceManager pm) {
         if (super.getTestList(pm) == null) {
             ArrayList<AbstractTest> list = new ArrayList<>();
-            list.add(new Experimental("stunreachability"));
-            list.add(new Experimental("dnscheck"));
-            list.add(new Experimental("torsf"));
-            list.add(new Experimental("vanilla_tor"));
+            if (getAutoRun()) {
+                list.add(new Experimental("torsf"));
+                list.add(new Experimental("vanilla_tor"));
+            } else {
+                list.add(new Experimental("stunreachability"));
+                list.add(new Experimental("dnscheck"));
+            }
```

Comment:
```suggestion
            if (getAutoRun()) {
                list.add(new Experimental("torsf"));
                list.add(new Experimental("vanilla_tor"));
            }
            list.add(new Experimental("stunreachability"));
            list.add(new Experimental("dnscheck"));
```

We should run all experiments in autorun mode and the just the slow ones otherwise.